src/js/providers/utils/quality-labels.ts

```diff
@@ -1,24 +1,36 @@
+import type { Level } from 'providers/data-normalizer';
+import type { GenericObject } from 'types/generic.type';
+
+type QualityLabels = {
+    [key: number]: string;
+};
+
 // Try and find a corresponding custom label. If there are no custom labels, create one using height, bandwidth, or both
-export function generateLabel(level, qualityLabels, redundant) {
+export function generateLabel(level: Partial<Level>, qualityLabels: QualityLabels | null, redundant: boolean): string {
```

Comment:
This is what an hls.js (v1) level (`hlsjs.levels[]`) looks like:

```js
export class Level {
  public attrs: LevelAttributes;
  public audioCodec?: string;
  public audioGroupIds?: string[];
  public bitrate: number;
  public details?: LevelDetails;
  public fragmentError: boolean = false;
  public height: number;
  public id: number;
  public loadError: number = 0;
  public loaded?: any;
  public name: string | undefined;
  public realBitrate: number = 0;
  public textGroupIds?: string[];
  public url: string[];
  public urlId: number = 0;
  public videoCodec?: string;
  public width: number;
  public unknownCodecs: string[] | undefined;

  constructor (data: LevelParsed) {
    this.url = [data.url];
    this.attrs = data.attrs;
    this.bitrate = data.bitrate;
    this.details = data.details;
    this.id = data.id || 0;
    this.name = data.name;
    this.width = data.width || 0;
    this.height = data.height || 0;
    this.audioCodec = data.audioCodec;
    this.videoCodec = data.videoCodec;
    this.unknownCodecs = data.unknownCodecs;
  }

  get maxBitrate (): number {
    return Math.max(this.realBitrate, this.bitrate);
  }
}
```

For shaka it's `shakaPlayer.getVariantTracks()[]` https://shaka-player-demo.appspot.com/docs/api/shaka.extern.html#.Track
(they also use `bandwidth` rather than `bitrate`)